### PR TITLE
fix: surface provider error body in embedded-agent chat errors

### LINKED ISSUE #1258: embedded-agent: surface the provider's error body in chat, not just the bare HTTP status
## Summary

When an OpenAI-compatible provider rejects a request, the embedded-agent chat surfaces only the bare HTTP status (`provider responded with HTTP 403`). The provider's structured error body — which typically names the actual cause — is read from the wire and discarded.

## Real incident that motivated this (owner dogfood, 2026-08-01)

During #1004 dogfood item 1, chat against `https://opencode.ai/zen/go/v1` with model `deepseek-v4-flash` failed with a bare 403. Diagnosing it required reproducing the exact `POST /chat/completions` with `curl`, which finally revealed:

```json
{"type":"error","error":{"type":"RegionError","message":"The latest version of this model is only available hosted in China and requires explicit opt in: https://opencode.ai/workspace/.../go"}}
```

The provider had answered the "why" precisely (model requires regional opt-in, with the fix URL included). None of it reached the UI. With the body surfaced, this would have been a zero-round-trip diagnosis instead of a multi-step curl session.

## Where the body is dropped

`packages/embedded-agent/src/providers/openai-chat-adapter.ts:172-177`:

```ts
if (!res.ok) {
  const retryable = res.status === 429 || res.status >= 500;
  throw new ProviderError(
    `provider responded with HTTP ${res.status}`,
    { retryable, status: res.status, retryAfterMs },
  );
}
```

The response body is never read on the error path.

## Expected behavior

- On `!res.ok`, read the body (bounded) and enrich the `ProviderError` message with the provider's own explanation before throwing, so it flows through the existing error event to the chat UI.
- Extraction heuristics (best-effort, never throwing from the error path itself):
  - If the body parses as JSON and has `error.message` (OpenAI shape) or `message`, use that (plus `error.type`/`error.code` when present).
  - Otherwise fall back to the raw text, truncated.
- **Bound the size** (e.g. first ~500 chars): WAF/gateway 403s are often full HTML pages; the UI needs the head of it, not 90KB of markup.
- Keep the existing `retryable` / `status` / `retryAfterMs` semantics unchanged.
- The API key must never appear in the surfaced message (it cannot, structurally — it lives in the request, not the response — but the test should assert the message construction never touches the auth header).

## Acceptance criteria

- [ ] A non-ok provider response with an OpenAI-shape JSON error body produces a `ProviderError` whose message includes the provider's `error.message` (and type/code when present) alongside the HTTP status; the chat UI error banner shows it.
- [ ] A non-ok response with a non-JSON body (e.g. HTML) surfaces the truncated head of the body.
- [ ] Bodies longer than the cap are truncated (test at the boundary).
- [ ] An unreadable body (stream error) degrades to today's status-only message — the error path never throws a secondary error.
- [ ] Polarity: the enriched-message assertions fail against the current body-discarding implementation.
- [ ] `retryable` classification and `retryAfterMs` behavior are unchanged (existing tests stay green).

## Out of scope

- Retry/backoff policy changes.
- Error surfacing for the MCP tool-call path (separate channel; audit later if a similar gap is observed).

Reversibility: reversible (error-message enrichment, no behavior change on success paths).

### PR DESCRIPTION
**Pull request by ms2sato**
## Summary

- `OpenAIChatAdapter.run()` discarded the response body on `!res.ok`, so the chat UI only ever showed a bare `provider responded with HTTP <status>`. The provider's own structured explanation (region gating, WAF block reason, etc.) never reached the user, forcing a manual curl round-trip to diagnose (see Issue for the incident that motivated this).
- On a non-ok response, the body is now read best-effort and used to enrich the `ProviderError` message before throwing:
  - OpenAI-shape JSON (`error.message`, `error.type`, `error.code`) is preferred; a top-level `message` field is the fallback JSON shape.
  - Non-JSON bodies fall back to the raw text.
  - The detail is capped at 500 chars with a `[truncated]` marker when cut, so a WAF's full HTML error page doesn't flood the UI.
  - Reading/parsing the body can never throw a secondary error — any failure degrades to today's status-only message.
- `retryable` / `status` / `retryAfterMs` construction is untouched.

## Architect involvement

AC (6 items) were pre-confirmed by the architect before delegation; no additional consultation was needed during implementation — the two "optional micro-item" suggestions from the architect were both accepted (see below), decided by the implementer per the issue's explicit "your call" framing.

## Optional micro-items (architect-flagged, implementer's call)

- (a) Empty-body boundary test (`!res.ok` + `''` body stays status-only) — **included**: it's the AC's own explicit boundary case and costs nothing.
- (b) `[truncated]` indicator on truncation — **included**: cheap, and directly useful for the diagnosis use case this issue exists for (a reader needs to know the message was cut, not read a silently-truncated sentence as complete).

## Polarity check (TDD)

7 new tests were written first and run against the unmodified body-discarding implementation:

- **4 failed for the right reason** (message was status-only, missing all body content):
  - `includes the provider error message and type/code from an OpenAI-shape JSON error body`
  - `surfaces the truncated head of a non-JSON (e.g. HTML gateway) error body`
  - `does not truncate a body exactly at the 500-char cap`
  - `truncates a body one char over the 500-char cap and marks it as truncated`
- **3 passed pre-fix as expected** (they guard pre-existing/boundary behavior the fix must *preserve*, not the fix's own new behavior):
  - `stays status-only when the body is empty`
  - `degrades to today's status-only message when reading the body rejects, without throwing a secondary error`
  - `never surfaces the Authorization header value in the enriched error message`

After the fix, all tests in the file pass (30 → 34 tests, 0 fail).

## Out of scope

- Retry/backoff policy — unchanged.
- MCP tool-call error path — untouched (separate channel per Issue).
- Issue #1259 (key-store failure UI) — a sibling issue, deliberately not mixed into this PR (per architect's 2-PR-separation ruling).

## Verification

`bun run typecheck && bun run test` (full monorepo suite), last portion + exit code:

```
@agent-console/shared         596 pass, 0 fail
@agent-console/integration     73 pass, 0 fail
@agent-console/server        3730 pass, 1 skip (pre-existing, unrelated memfs-isolation skip), 0 fail
@agent-console/embedded-agent  294 pass, 0 fail
@agent-console/client         2107 pass, 0 fail
scripts/hooks/skills tests    521 pass, 0 fail
TEST_EXIT: 0
```

`node .claude/skills/orchestrator/preflight-check.js` — clean (coverage, rule/skill duplication, language, blame-shift checks all pass).

**CodeRabbit CLI**: could not authenticate in this headless worktree session (`coderabbit auth login` requires a user-controlled browser flow; automatic login timed out). Skipped per `workflow.md` fallback — recommend the GitHub-side CodeRabbit bot review be checked before merge.

## Update: addressed CodeRabbit Major finding (commit `dca4bf67`)

The GitHub-side CodeRabbit bot flagged a real Major issue on the first commit: `await res.text()` buffered the **entire** error-response body in memory before the 500-char display truncation applied — i.e. the read itself was unbounded even though display was bounded, directly contradicting this PR's own "don't flood the UI with a 90KB WAF page" rationale. A hostile or misbehaving gateway could force an unbounded allocation on the error path.

Fix: replaced `res.text()` with a bounded stream read (`readBoundedBodyText`, `MAX_PROVIDER_ERROR_BODY_BYTES = 8 * 1024`) that reads at most 8KB from `res.body` via `getReader()` and cancels the reader once the bound is hit. 8KB safely exceeds the 500-char display cap (so JSON `error.type`/`error.code` wrappers around `message` aren't cut before parsing) while still bounding well below "arbitrarily large". The existing best-effort contract (no secondary error from the enrichment path) is unchanged — reader errors still propagate into the same outer `try/catch`.

Added a regression test (`stops reading once the bounded prefix is reached, never consuming a much larger body`) that feeds a 24KB stream (3x the bound) and asserts the reader only pulls up to the bound before stopping — this test fails against the original `res.text()`-based implementation, which would have drained the whole stream. All existing enrichment-message tests (migrated from a `text` mock field to real `ReadableStream` bodies, since production no longer calls `.text()`) stay green.

Closes #1258

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error messages for provider request failures by including relevant response details.
  * Added safe handling for structured, plain-text, empty, or unreadable error responses.
  * Limited included error details to 500 characters while preserving status and retry information.
  * Redacted authorization data from surfaced error messages.

### CODE REVIEW

File: packages/embedded-agent/src/providers/openai-chat-adapter.ts [hunk 1]
```diff
@@ -172,10 +221,17 @@ export class OpenAIChatAdapter implements ProviderAdapter {
...
+        let message = `provider responded with HTTP ${res.status}`;
+        try {
+          const detail = extractProviderErrorDetail(await res.text());
+          if (detail !== undefined) {
+            message = `${message}: ${detail}`;
+          }
+        } catch {
+          // Enrichment is best-effort only; an unreadable body must not
+          // prevent throwing the status-only ProviderError below.
+        }
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply packages/embedded-agent/src/providers/openai-chat-adapter.ts:262]: Fixed in dca4bf67: replaced `res.text()` with a bounded stream read (`readBoundedBodyText`, 8KB cap via `res.body.getReader()` + `reader.cancel()` once the bound is hit). Added a regression test that feeds a 24KB stream and asserts the reader stops well short of draining it — the test fails against the original unbounded implementation. Display-side 500-char truncation is unchanged.
